core/runtime/ops.rs

```diff
@@ -119,38 +113,32 @@ pub fn queue_async_op<'s>(
   //   Some(scope.get_current_context())
   // );
 
-  // All ops are polled immediately
-  let waker = noop_waker();
-  let mut cx = Context::from_waker(&waker);
-
-  // Note that MaybeDone returns () from the future
-  let op_call = match op.poll_unpin(&mut cx) {
-    Poll::Pending => {
-      let MaybeDone::Future(fut) = op else {
-        unreachable!()
-      };
-      OpCall::pending(ctx, promise_id, fut)
-    }
-    Poll::Ready(_) => {
-      let mut op_result = Pin::new(&mut op).take_output().unwrap();
-      // If the op is ready and is not marked as deferred we can immediately return
-      // the result.
-      if !deferred {
-        ctx.state.borrow_mut().tracker.track_async_completed(ctx.id);
-        return Some(op_result.to_v8(scope).unwrap());
+  let id = ctx.id;
+  if deferred {
+    ctx
+        .context_state
+        .borrow_mut()
+        .pending_ops
+        // SAFETY: this this is guaranteed to be running on a current-thread executor
```

Comment:
Super tiny nit: "this this" should probably be only one "this"

Reply:
this this is a good point! 😆 fixed